### PR TITLE
Update note about MacOS compatibility

### PR DESCRIPTION
**Pull request by jnmclarty**
## What changes are proposed in this pull request?

The note about compatibility with MacOS may be unnecessarily harsh or decoupled from reality.  If it's not, I'd like to better understand what is and what isn't compatible with MacOS, as soon as possible.

## How is this patch tested?

It wasn't. It's just a change to docs.  Maintainers may need to confirm the docs build and format correctly.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [X] Examples
- [ ] Docs
- [X] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by jnmclarty** [comment]
Mostly proposing this PR, just to poke at understanding compatibility of mlflow.  The note as is, almost steered me away from even trying to use it.

File: docs/source/quickstart.rst [hunk 1]
```diff
@@ -21,9 +21,11 @@ You install MLflow by running:
...
 .. note::
 
-    You cannot install MLflow on the MacOS system installation of Python. We recommend installing
-    Python 3 through the `Homebrew <https://brew.sh/>`_ package manager using
-    ``brew install python``. (In this case, installing MLflow is now ``pip3 install mlflow``).
+    Certain parts of MLflow do not function on the MacOS system due to installation issues related to Python. We
+    recommend installing Python 3 through the `Homebrew <https://brew.sh/>`_ package manager using
+    ``brew install python``. (In this case, installing MLflow is now ``pip3 install mlflow``).  
+    
+    As of 1.6, mlflow tracking is easy to install and works fine on MacOS.
```
Comment (aarondav) [inline docs/source/quickstart.rst:28]:
```suggestion
    MLflow works on MacOS. If you run into issues with the default system Python on MacOS, try
    installing Python 3 through the `Homebrew <https://brew.sh/>`_ package manager using
    ``brew install python``. (In this case, installing MLflow is now ``pip3 install mlflow``).  
```

File: docs/source/quickstart.rst [hunk 2]
```diff
@@ -21,9 +21,9 @@ You install MLflow by running:
 
 .. note::
 
-    You cannot install MLflow on the MacOS system installation of Python. We recommend installing
-    Python 3 through the `Homebrew <https://brew.sh/>`_ package manager using
-    ``brew install python``. (In this case, installing MLflow is now ``pip3 install mlflow``).
+    MLflow works on MacOS. If you run into issues with the default system Python on MacOS, try
+    installing Python 3 through the `Homebrew <https://brew.sh/>`_ package manager using
+    ``brew install python``. (In this case, installing MLflow is now ``pip3 install mlflow``).
```
Comment (aarondav) [inline docs/source/quickstart.rst:26]:
```suggestion
    ``brew install python``. (In this case, installing MLflow is now ``pip3 install mlflow``).
```

**Comment by aarondav** [comment]
Agree the current note is a bit heavy-handed. I updated it to be even lighter, just a suggestion if people run into issues.

**Review by aarondav — APPROVED** [review]

**Pull request merged**